File: 9. Palindrome Number/is_palindrome.ts
function isPalindrome(x: number): boolean {
  let s: string = `${x}`;
  let start: number = 0;
  let finish: number = s.length - 1;

  while(start !== finish && start < finish) {
    if (s[start] === s[finish]) {
      start++;
      finish--;
    } else {
      return false
    }
  }
  return true;
};

function isPalindrome2(x: number): boolean {
  let s: string = `${x}`;
  return s === s.split('').reverse().join('')
};


function isPalindrome3(x: number): boolean {
  let s: string = `${x}`;
  let s2: string = '';
  for (let char of s) s2 = char + s2;

  return s === s2;
};


function isPalindrome4(x: number): boolean {
  let s: string = `${x}`;
  
  return s === Array.prototype.reduceRight.call(s, (acc, n) => acc + n, '')
};


console.log(isPalindrome(1111));
console.log(isPalindrome(121));
console.log(isPalindrome(-121));
console.log(isPalindrome(10));
console.log(isPalindrome(-101));

File: 9. Palindrome Number/is_palindrome.rb
def is_palindrome(x)
  "#{x}" == "#{x}".reverse
end

def is_palindrome(x)
  s = "#{x}"
  start = 0
  finish = s.size - 1

  while start != finish && start < finish
    if s[start] == s[finish]
      start += 1
      finish -= 1
    else return false
    end
  end
  
  true
end

pp is_palindrome 121
pp is_palindrome 11111
pp is_palindrome -121
pp is_palindrome 10
pp is_palindrome -101


